src/components/LazyLog/index.jsx

```diff
@@ -327,6 +346,75 @@ export default class LazyLog extends Component {
     });
   };
 
+  handleSearch = keywords => {
+    const { encodedLog, resultLines, searchKeywords } = this.state;
+    let currentResultLines = resultLines;
+
+    if (keywords !== searchKeywords) {
+      currentResultLines = searchLines(keywords, encodedLog);
+    }
+
+    this.setState({
+      resultLines: currentResultLines,
+      isSearching: true,
+      searchKeywords: keywords,
+    });
+  };
+
+  handleClearSearch = () => {
+    this.setState({
+      isSearching: false,
+      searchKeywords: '',
+      resultLines: [],
+      filteredLines: List(),
+      resultLineUniqueIndexes: [],
+      isFilteringLinesWithMatches: false,
+      scrollToIndex: 0,
+    });
+  };
+
+  handleFilterLinesWithMatches = isFilteringLinesWithMatches => {
+    const { resultLines, lines } = this.state;
+
+    this.setState({
+      isFilteringLinesWithMatches,
+      filteredLines: List(),
+      resultLineUniqueIndexes: [],
+    });
+
+    if (resultLines.length > 0 && isFilteringLinesWithMatches) {
+      const resultLineUniqueIndexes = [...new Set(resultLines)];
+
+      this.setState({
+        resultLineUniqueIndexes,
+        filteredLines: lines.filter((line, index) =>
+          resultLineUniqueIndexes.some(
+            resultLineIndex => index + 1 === resultLineIndex
+          )
+        ),
+      });
+    }
+  };
+
+  handleFormatPart = () => {
+    const { formatPart } = this.props;
+    const { isSearching, searchKeywords } = this.state;
+
+    if (isSearching) {
+      return searchFormatPart({
```

Comment:
This returns a function?

Reply:
Yes, it returns the function in https://github.com/ndiaz-candido/react-lazylog/blob/118e3f86330522af2dd4a720e8fc9512315712c7/src/utils.js#L114 which is the expected for `Line.formatPart` prop.